addons/hr_holidays/static/src/js/time_off_calendar.js

```diff
@@ -11,6 +12,15 @@ odoo.define('hr_holidays.dashboard.view_custo', function(require) {
     var _t = core._t;
     var QWeb = core.qweb;
 
+    var TimeOffCalendarModel = CalendarModel.extend({
+
+        _getFilterDomain: function() {
+            const company_domain = [['user_id.company_id', 'in', this.data.context.allowed_company_ids]];
```

Comment:
This should have been `['employee_company_id', 'in', this.data.context.allowed_company_ids]` instead